### ISSUE TITLE
Add tlsConfig for ingester ring consul and etcd kvstore

### ISSUE THREAD

**Issue by reidlai** [grafana/loki#2428]
**Is your feature request related to a problem? Please describe.**
Based on current Loki configuration, there is no options to provide ca.crt, tls.crt and tls.key if Consul and Etcd are using HTTPS for connection

**Describe the solution you'd like**
if there is option tlsConfig and allow us to specify path or secret name which store ca.crt, tls.crt, and tls.key, it will be easy to implement private inter pod connection.
 
**Describe alternatives you've considered**

**Additional context**

**Comment by miro-balaz** [comment]
Maybe mounting CA file into standard location would work [](https://golang.org/src/crypto/x509/root_linux.go)

**Comment by tcassaert** [comment]
Could this be reopened? This currently makes it impossible to use the consul kv store when you have your consul setup with TLS.

**Comment by mxab** [comment]
We're facing the same issue currently.

I looked a bit in the code.
As far as I unstand this Loki uses [Grafana's Distributed systems kit](https://github.com/grafana/dskit)

For consul it uses the official consul client, which can already be configured a lot by environment variables.

Unfortunatelly the way they instanciate the client the tls related env vars are irgnored.

I created an issue for this here:
https://github.com/grafana/dskit/issues/346

and a potential fix for it https://github.com/grafana/dskit/pull/348
